packages/contracts/test/contracts/discovery/Batch.ts

```diff
@@ -0,0 +1,49 @@
+describe("Batch", () => {
+  describe("constructor", () => {
+    it("works with valid arguments");
+    it("fails with an empty list of projects");
+    it("fails with an address that's not a project");
+    it("fails if one of the projects is not whitelisted");
+  });
+
+  describe("vote", () => {
+    it("accepts votes for projects");
+    it("does not accept two votes from the same user in the same project");
+    it(
+      "does not accept votes from people who exhausted their votes for this batch"
```

Comment:
How many votes do they have per batch? Is this something the controller will decide?

Reply:
yes